### DISCUSSION TITLE
Cannot train model, TypeError: '>' not supported between instances of 'NoneType' and 'int'

### DISCUSSION THREAD

**Discussion by ginbobby** [talmolab/sleap#1852] [Help!] [answered]
Hello! I realize this issue has been reported before, but I cannot fix it.

I previously was able to train my model just fine, but now when I try and train, I get this error in the console:
TypeError: '>' not supported between instances of 'NoneType' and 'int'

It's using the same videos and I have trained fine in this project before, plus file path was unchanged. Since I figured it was an issue with the path, I re-imported all of my videos to a new folder and replaced the videos in sleap. But this did not work, I still get this popup:
"An error occurred while training centroid/centered_instance. Your command line terminal may have more information about the error."

Any help would be great! I've never used python before, so I don't really know what is happening half the time.

![image](https://github.com/talmolab/sleap/assets/170107232/70d4e3d2-ed01-439b-a19e-ac393ff54547)

**Comment by gitttt-1234** [comment, marked as answer]
Hi @ginbobby !

From the error, the crash is in `providers.py` at `max_height_and_width` where it's trying to compute `max(video.shape[1] for video in self.videos)`, and one of the videos has a `None` shape. This can happen when a video file can't be opened (path not resolvable, file moved/renamed), or if the project was saved when a video's shape couldn't be determined. 

Since this was posted, I/O has moved to [**sleap-io**](https://io.sleap.ai/latest/) and training has moved to [**sleap-nn**](https://nn.sleap.ai/latest/) (the backend SLEAP uses for all deep learning). Could you [update SLEAP to v1.6.3](https://docs.sleap.ai/latest/installation/)? 

Closing this in favor of the updated training pipeline, feel free to reopen or start a new discussion if you hit this again!
